### PR TITLE
docs: add PR template

### PR DESCRIPTION
**Pull request by subhamkrai**

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+Thank you for contributing to Rook! -->
+
+**Description of your changes:**
+
+**Which issue is resolved by this Pull Request:**
+Resolves #
+
+**Checklist:**
+
+- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
```
Comment (travisn) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: If we're going to ask for the same structure, let's enable the commitlint bot and copy over the same .commitlintrc.json to get started.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
+- [ ] CI tests has been updated, if necessary.
```
Comment (travisn) [inline .github/PULL_REQUEST_TEMPLATE.md:15]: How are CI tests different than unit tests and integration tests? They have separate bullets below.

**Review by travisn — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (subhamkrai) [reply .github/PULL_REQUEST_TEMPLATE.md:15]: right, remove the integration tests section as CI tests make more sense with tool.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (subhamkrai) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: copied from rook and trimmed the commit title list.

**Review by travisn — APPROVED** [review]

**Comment by subhamkrai** [comment]
Merging since CI issue in not related

**Pull request merged**